js/coinex.js

```diff
@@ -3879,6 +3880,95 @@ module.exports = class coinex extends Exchange {
+            request['market'] = market['id'];
+        }
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        const response = await this.privateGetMarginLoanHistory (this.extend (request, params));
+        //
+        //     {
+        //         "code": 0,
+        //         "data": {
+        //             "page": 1,
+        //             "limit": 10,
+        //             "total": 1,
+        //             "has_next": false,
+        //             "curr_page": 1,
+        //             "count": 1,
+        //             "data": [
+        //                 {
+        //                     "loan_id": 2616357,
+        //                     "create_time": 1654214027,
+        //                     "market_type": "BTCUSDT",
+        //                     "coin_type": "BTC",
+        //                     "day_rate": "0.001",
+        //                     "loan_amount": "0.0144",
+        //                     "interest_amount": "0",
+        //                     "unflat_amount": "0",
+        //                     "expire_time": 1655078027,
+        //                     "is_renew": true,
+        //                     "status": "finish"
+        //                 }
+        //             ],
+        //             "total_page": 1
+        //         },
+        //         "message": "Success"
+        //     }
+        //
+        const data = this.safeValue (response, 'data', {});
+        const rows = this.safeValue (data, 'data', []);
+        const interest = this.parseBorrowInterests (rows, market);
+        return this.filterByCurrencySinceLimit (interest, code, since, limit);
+    }
+
+    parseBorrowInterest (info, market = undefined) {
+        //
+        //     {
+        //         "loan_id": 2616357,
+        //         "create_time": 1654214027,
+        //         "market_type": "BTCUSDT",
+        //         "coin_type": "BTC",
+        //         "day_rate": "0.001",
+        //         "loan_amount": "0.0144",
+        //         "interest_amount": "0",
+        //         "unflat_amount": "0",
+        //         "expire_time": 1655078027,
+        //         "is_renew": true,
+        //         "status": "finish"
+        //     }
+        //
+        const marketId = this.safeString (info, 'market_type');
+        market = this.safeMarket (marketId, market);
+        const symbol = this.safeString (market, 'symbol');
+        const timestamp = this.safeTimestamp (info, 'expire_time');
+        const unflatAmount = this.safeString (info, 'unflat_amount');
+        const loanAmount = this.safeString (info, 'loan_amount');
+        let interest = Precise.stringSub (unflatAmount, loanAmount);
+        if (unflatAmount === '0') {
+            interest = undefined;
+        }
+        return {
+            'account': undefined, // deprecated
+            'symbol': symbol,
+            'marginMode': undefined,
+            'marginType': undefined, // deprecated
```

Comment:
Find out if coinex has both cross and/or isolated margin, if they only have one, then this field will be that type, if they have both, you can figure out which type this should be usually with some tricks, such as, with isolated margin, a market symbol would always be required, but a symbol would not be required for cross margin, cross and isolated margin wallets will usually have different properties in the response, like isolated wallets should probably always have the market symbol in the response while cross might not

Reply:
Coinex has cross and isolated for perpetual futures trading but they handle margin trading a bit differently than most exchanges, you need to deposit funds to the specific margin trading pair before you can borrow or trade on that pair so it looks like all margin trading positions are isolated